client/asset/eth/eth.go

```diff
@@ -1454,11 +1454,12 @@ func (w *assetWallet) swapGas(n int, ver uint32) (oneSwap, nSwap uint64, approve
 	// If a live estimate is greater than our estimate from configured values,
 	// use the live estimate with a warning.
 	if gasEst, err := w.estimateInitGas(w.ctx, n, ver); err != nil {
-		w.log.Errorf("(%d) error estimating swap gas: %v", w.assetID, err)
+		w.log.Errorf("(%d) error estimating swap gas (using expected gas cap instead): %v", w.assetID, err)
 		// TODO: investigate "gas required exceeds allowance".
-		return 0, 0, false, err
+		// TODO: investigate "error getting gas fees: execution reverted: transfer from failed"
+		// return 0, 0, false, err
```

Comment:
I believe whether we fail depends on the caller.  `FundOrder` should probably fail, whilst `Swap` should not.
This will take some refactoring unfortunately.

Reply:
Proposed solution in https://github.com/decred/dcrdex/pull/2118/commits/6101a6cb554437df45c334aef7dcbe3d5593e949